### PR TITLE
✅ Add mcdoc runtime tests

### PR DESCRIPTION
**Pull request by misode**
Depends on #1187

### CODE REVIEW

File: packages/mcdoc/test/runtime/checker.spec.ts [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+			return { kind: 'any' }
+		}
+	}
+
+	for (const { name, type, values, init } of suites) {
+		describe(`typeDefinition ${localeQuote(name)}`, () => {
+			for (const value of values) {
+				it(`with value ${JSON.stringify(value)}`, () => {
+					const errors: McdocCheckerError<JsValue>[] = []
+					const project = mockProjectData()
```
Comment (NeunEinser) [inline packages/mcdoc/test/runtime/checker.spec.ts:392]: This looks useful!

**Review by NeunEinser — APPROVED** [review]
This looks nice for a start. I'd probably want to write some more in-depth tests for some things interacting in weird ways, but this is good for a start.

I think it still makes sense if I write some tests too, as I have thought _a lot_ about all kinds of weird edge cases in the past week, and I am probably the most into it rn :)

But yeah we should merge this as soon as possible. Thanks for your help, really appreciate it :) <3

File: packages/mcdoc/test/runtime/checker.spec.ts (hunk 1 above)
Comment (misode) [reply packages/mcdoc/test/runtime/checker.spec.ts:392]: Yep, it's a pretty useful utility
